src/Sarif/Visitors/MakeUrisAbsoluteVisitor.cs

```diff
@@ -19,19 +19,33 @@ public override FileLocation VisitFileLocation(FileLocation node)
         {
             if ( _run.OriginalUriBaseIds!= null &&
                 !string.IsNullOrEmpty(node?.UriBaseId) &&
-                _run.OriginalUriBaseIds.ContainsKey(node.UriBaseId))
+                _run.OriginalUriBaseIds.ContainsKey(node.UriBaseId) && 
+                !_run.OriginalUriBaseIds.Values.Contains(node))
             {
-                Uri baseUri = _run.GetExpandedUriBaseIdValue(node.UriBaseId);
+                Uri baseUri = _run.ExpandUrisWithUriBaseId(node.UriBaseId);
                 node.Uri = CombineUris(baseUri, node.Uri);
                 node.UriBaseId = null;
             }
 
             return node;
         }
 
-        private Uri CombineUris(Uri baseUri, Uri uri)
+        internal static Uri CombineUris(Uri absoluteBaseUri, Uri relativeUri)
         {
-            throw new NotImplementedException();
+            if (!absoluteBaseUri.IsAbsoluteUri)
+            {
+                throw new ArgumentException("absoluteBaseUri is not an absolute Uri", nameof(absoluteBaseUri));
+            }
+
+            if (relativeUri.IsAbsoluteUri)
+            {
+                throw new ArgumentException("relativeUri is not a relative Uri", nameof(relativeUri));
+            }
+
+            Uri test = new Uri("src/archive.zip#archive2.gz", UriKind.Relative);
+
```

Comment:
Oops :-)

Reply:
Fixed.

---
In reply to: [247331755](https://github.com/Microsoft/sarif-sdk/pull/1210#discussion_r247331755) [](ancestors = 247331755)